lib/components/viewers/RouteRow.js

```diff
@@ -138,26 +183,29 @@ export class RouteRow extends PureComponent {
     }
   }
 
-  // Used to ensure details are visible until animation completes
-  setNewHeight = (newHeight) => {
-    this.setState({ newHeight })
+  _patternButtonClicked = () => {
+    const { route, setViewedRoute } = this.props
+    const { id, patterns } = route
+    const firstPattern = route && patterns && Object.values(patterns)[0]?.id
```

Comment:
```suggestion
    const firstPattern = route && patterns && Object.values(patterns)?.[0]?.id
```

Can we guarantee that patterns will be a non-empty object?

Reply:
Good call!